### ISSUE TITLE
How to run *all* tests?

### ISSUE THREAD

**Issue by cfjedimaster** [lirantal/swagger-lint-api#1]
Pardon me if I'm missing the obvious, but the sample code shows:

```
const result = validator.descriptionHasNoLineBreaks();
```

which works fine, but how would I iterate over all the validator methods and run them all? (Outside of hard coding of course. :)

**Comment by lirantal** [comment]
I just started working on this so this is very early, but then again, it means great room for improvement and for Pull Requests! 😀🤗 yay!

The lib is currently as simple as invoking every method but I wanted to build a CLI tool around it to do exactly what you're saying. Had actually shared my thoughts on twitter https://twitter.com/liran_tal/status/1150471462589009922 and tagged you now to so we can follow-up on it.

Generally speaking we can tweak up the class to return itself, and so we can chain methods. Happy to hear more thoughts on this :)

**Comment by cfjedimaster** [comment]
Honestly I'm not sure. You could hard code it and then it becomes "your" problem such that when you add a new validator you remember to update the list. I think that would be an acceptable quick fix. You could also maybe use Reflection (ok, that's a java thing but you get my meaning) to see if you can inpsect the methods of the class and sniff for methods that begin with 'description'.

**Comment by lirantal** [comment]
Let's talk from the a user perspective. What kind of API would you want?

Imagine you'd be able to write your expectations like in tests for your swagger file.
Would something like this work?
```js
expect(mySwaggerJSON, 'not to have line-breaks');
expect(mySwaggerJSON, 'not to have tabs');
```

**Comment by cfjedimaster** [comment]
I was more thinking:

```
const result = validator.executeAll();
```

Result is an array of issues, with the Result object being decorated with something to let me know the rule that threw it:

```
{ 
rule: 'nolinebreaks',
valid: false,
  path:
   [ 'definitions',
     'SessionTokenErrorMessage',
     'properties',
     'code',
     'description' ] }
```

**Comment by lirantal** [comment]
Uhmm. We could generally expose something like this but that would be by internally having that method call each of the others explicitly. I don't mind doing that, however some of the method calls don't make sense.

For example, we have `DescriptionValidator.descriptionEndsWithString(string)` which expects an argument (the string to match). Also the method call `DescriptionValidator.descriptionCompliesWithFunction(fn)` is a custom function invocation. Both of these don't make sense for an `executeAll()`.

So it comes to a matter of perhaps exposing an  "opinionated" `executeAll that only invokes the straight-forward method calls, or... maybe the user consuming the lib builds that per their own needs.

TBH I don't have a strong opinion.

**Comment by cfjedimaster** [comment]
Yeah honestly I'm less interested in this now versus adding support for multiple results. Going back to that bug to comment. :)

**Comment by lirantal** [comment]
Hah cool! Let's see how to work on those multiple results.

**Comment by lirantal** [comment]
Just to put here as a future reference that I think [unexpected](https://unexpected.js.org/api/addAssertion) could be a nice way to wrap this functionality into an eslint kind of rules.
